mpmath/functions/hypergeometric.py

```diff
@@ -1062,6 +1063,82 @@ def h(*args):
             return terms
     return ctx.hypercomb(h, a+b, **kwargs)
 
+@defun
+def foxh(ctx, aA_s, bB_s, z, r=1, series=None, **kwargs):
+    aAn, aAp = aA_s
+    bBm, bBq = bB_s
+    n = len(aAn)
+    p = n + len(aAp)
+    m = len(bBm)
+    q = m + len(bBq)
+    aA = aAn+aAp
+    bB = bBm+bBq
+    a = [ctx.convert(a) for a, _ in aA]
+    b = [ctx.convert(b) for b, _ in bB]
+    A = [A for _, A in aA]
+    B = [B for _, B in bB]
+    z = ctx.convert(z)
+    r = ctx.convert(r)
+
+    # Convert integer to rationals in form Ai / 1
+    Afrac = [(Ai, 1) if isinstance(Ai, int) else Ai for Ai in A]
+    Bfrac = [(Bj, 1) if isinstance(Bj, int) else Bj for Bj in B]
+
+    if not all(isinstance(c, int) and isinstance(d, int) and
+               c > 0 and d > 0 for c, d in Afrac + Bfrac):
+        raise NotImplementedError("All A and B must be positive rationals")
```

Comment:
That's not true, you are accepting only integers, right?
BTW, mpf's ARE rationals, i.e. you can use MeijerG identify in general.  Though, not sure it worth.

Reply:
I've adapted the code to accept either `A = n` an integer or `A = (p,q)` a rational `p/q`; the code above converts `A = n` to `A = (n, 1)`, and this line checks the condition that they are all in form `(p, q)`.

While mpf are rationals, I think letting user specify `(p,q)` instead of `mpf('p/q')` has the behefit that I could recover the true denominator more easily, and LCM would be small; I am not sure `mpf('p/q').to_rational` would exactly return `q` for me, if it is, I could change A to accept mpf as well.

Reply:
> I've adapted the code to accept either A = n an integer or A = (p,q) a rational p/q; the code above converts A = n to A = (n, 1)

I see.  No, this is not good.  I don't think we have other function, accepting rationals that way.  Every parameter should be just some number.

> While mpf are rationals, I think letting user specify (p,q) instead of mpf('p/q') has the behefit that I could recover the true denominator more easily

Just as for a builtin float, you can use `as_integer_ratio()` method to recover numerator/denominator for a mpf.

Reply:
> I don't think we have other function, accepting rationals that way. Every parameter should be just some number.

I found such usage here https://mpmath.org/doc/current/functions/zeta.html

>  The parameter a is usually a rational number `p/q`, and may be specified as such by passing an integer tuple `(p,q)`.

Anyway, I'll change it to use mpf